acoustid/indexclient.py

```
@@ -87,20 +87,21 @@ def _connect(self):
                 % (self.host, self.port, e)
             )
 
-    def _putline(self, line):
-        # type: (str) -> None
+    def _putline(self, line: str) -> None:
+        assert self.sock is not None
         request = b"%s%s" % (line.encode("utf8"), CRLF)
         logger.debug("Sending request %r", request)
         self.sock.sendall(request)
 
     def _getline(self, timeout=None):
```

Ruff (0.8.2)

`_getline` is too complex (13 > 10)

(C901)